symengine/sets.h

```diff
@@ -249,6 +249,39 @@ class Complement : public Set
     }
 };
 
+class ConditionSet : public Set
+{
+private:
+    vec_sym syms_;
```

Comment:
for allowing it to be `FiniteSet`, `RCP<const Basic>` should be the data member right instead of `RCP<const Symbol>`?
We can currently define it only if the input is of type Symbol.